### PR TITLE
Add dry command and add semantic release plugins config

### PR DESCRIPTION
**Pull request by hobbitronics**
## Added
- Added dry command to see notes and version from semantic-release (see partial output below)
- Added release.config.js (copied from https://blog.elantha.com/semantic-release-local-dry-run/)

## Changed
- release-config: use conventionalcommits and custom release rules and notes with semantic release

Note: I had to merge main into this branch to not get all the changes since version 3.2

```[10:45:32 AM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[10:45:32 AM] [semantic-release] › ℹ  The next release version is 10.1.0
[10:45:32 AM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[10:45:32 AM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[10:45:32 AM] [semantic-release] › ⚠  Skip v10.1.0 tag creation in dry-run mode
[10:45:32 AM] [semantic-release] › ✔  Published release 10.1.0 on default channel
[10:45:32 AM] [semantic-release] › ℹ  Release note for version 10.1.0:
## 10.1.0 (https:/Users/mike/projects/ui-components//compare/v10.0.1...v10.1.0) (2023-03-28)

### Changed

    * release-config: use conventionalcommits and custom release rules and notes with semantic release (3537856 (https:/Users/mike/projects/ui-components//commit/3537856be3d599a89a6b054039c3965eb09b9c19))```

### CODE REVIEW

File: release.config.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        },
+        {
+          type: 'remove',
+          release: 'minor',
+        },
+        {
+          type: 'security',
+          release: 'patch',
+        },
+        {
```
Comment (hobbitronics) [inline release.config.js:33]: Maybe we don't even need the types below unless we want to use one of them to not show up in the changelog

File: release.config.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+          },
+          {
+            type: 'remove',
+            section: 'Removed',
+          },
+          {
+            type: 'security',
+            section: 'Security',
+          },
+          {
```
Comment (hobbitronics) [inline release.config.js:97]: same as above

File: release.config.js (hunk 1 above)
Comment (hobbitronics) [reply release.config.js:33]: I tried `release: false` but that caused an error

**Review by Schparky — APPROVED** [review]

File: release.config.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+          type: 'feat',
+          release: 'minor',
+        },
+        {
+          type: 'fix',
+          release: 'patch',
+        },
+        {
+          type: 'change',
+          release: 'patch',
```
Comment (hobbitronics) [inline release.config.js:23]:
```suggestion
          release: 'minor',
```

**Pull request merged**